### DISCUSSION TITLE
What is the best way to spawn two actors that talks to each other?

### DISCUSSION THREAD

**Discussion by Escapingbug** [slawlor/ractor#212] [Q&A] [answered]
I have a situation that needing two actors talking each other.
Originally I was thinking about saving ActorRefs in two actors, which seems to be a problem as no one could be spawned since they both need another one to be spawned first.
My current thought is to use the `OutputPort`, so that one could use the output port to spin up first then be passed to another one.
The problem is, I'm not sure the `OutputPort` is exactly designed for this purpose, and this seems to be a little bit low-level as I need to do some pub-sub preparation works for this situation.
Is there better way to achieve this?

**Comment by slawlor** [comment, marked as answer]
Generally it's not super easy to set up two actors with references to each other since that forms a cycle. What normally you would do start actor a, then start actor b passing a handle to a in B's startup routine. You and then send a message to actor a that b is running, and past the handle to b. Then they can send messages to each other.

However if you only need one actor to call the other and get a reply, that's what RPC is for. If you have a toy sample or a gist I could maybe give more guidance

**Comment by cerceris** [comment]
You don't need to pass references / handlers. The child actor uses the passed with init attributes parent actor name. The parent actor knows the child names because it created it. "name" = any actor ID that you use in your system.
